static/extensions/MubiLop/penguingpt.js

```diff
@@ -0,0 +1,752 @@
+                        }]
+                    },
+
+                    igModels: {
+                        acceptReporters: true,
+                        items: [
+                        {
+                            text: "DALL-E 3",
+                            value: "dalle-3"
+                        }, 
+                        {
+                            text: "Midjourney",
+                            value: "midjourney"
+                        },
+                        {
+                            text: "OpenJourney V4",
+                            value: "openjourney-v4"
+                        }, 
+                        {
+                            text: "Dreamshaper 8",
+                            value: "dreamshaper-8"
+                        }, 
+                        {
+                            text: "Anything V5",
+                            value: "anything-v5"
+                        }, 
+                        {
+                            text: "Realistic Vision V5",
+                            value: "realistic-vision-v5"
+                        }]
+                    },
+
+                    reqModels: {
+                        acceptReporters: true,
+                        items: [
+                        {
+                            text: "GPT 3.5 Turbo",
+                            value: "gpt-3.5-turbo"
+                        }, 
+                        {
+                            text: "GPT 4",
+                            value: "gpt-4"
+                        }, 
+                        {
+                            text: "GPT 4 1066 Preview (Recent)",
+                            value: "gpt-4-1106-preview"
+                        }, 
+                        {
+                            text: "GPT 4 0125 Preview (Turbo)",
+                            value: "gpt-4-0125-preview"
+                        }, 
+                        {
+                            text: "Llama 2",
+                            value: "llama-2-7b"
+                        }]
+                    }
+
+                }
+            };
+        }
+
+        getPrompt(args) {
+            if (args.TYPE !== '(select a prompt)') {
+                return args.TYPE;
+            } else {
+                return '';
+            }
+        }
+
+        setModel(args) {
+            this.model = args.MODEL
+        }
+
+        setApiUrl(args) {
```

Comment:
`api_url` is not respected for a lot of endpoints, this should either be removed or custom api url should be supported on endpoints that are hardcoded to reverse.mubi.tech